lib/ansible/plugins/strategy/__init__.py

```diff
@@ -530,6 +530,8 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
                                 for listening_handler_block in iterator._play.handlers:
                                     for listening_handler in listening_handler_block.block:
                                         listeners = getattr(listening_handler, 'listen', []) or []
+                                        if not isinstance(listeners, list):
```

Comment:
i would specifically check for 'string_types'  and then if its not list, raise an error.

Reply:
`listen = FieldAttribute(isa='list', default=list)` <= i would also check why this is not working as intended

Reply:
Good thought. As a result of consolidating the handler tracking dictionaries suddenly became viable too
```
listen:
  handler: new_workflow
```
this fixes that now